cmd/simulator/load/loader.go

```diff
@@ -27,15 +24,101 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/log"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"golang.org/x/sync/errgroup"
 )
 
 const (
 	MetricsEndpoint = "/metrics" // Endpoint for the Prometheus Metrics Server
 )
 
+type Loader[T txs.THash] struct {
+	clients     []txs.Worker[T]
+	txSequences []txs.TxSequence[T]
+	batchSize   uint64
+	metrics     *metrics.Metrics
+}
+
+func New[T txs.THash](
```

Comment:
NewLoader maybe?

Reply:
We've started explicitly removing names like this that repeat the package name, so I'd prefer to continue that trend and use `New` here so that the caller uses `load.New(...)` instead of `load.NewLoader(...)`